src/wallet/wallet2.cpp

```diff
@@ -1005,9 +1005,23 @@ gamma_picker::gamma_picker(const std::vector<uint64_t> &rct_offsets, double shap
   const size_t blocks_in_a_year = 86400 * 365 / DIFFICULTY_TARGET_V2;
   const size_t blocks_to_consider = std::min<size_t>(rct_offsets.size(), blocks_in_a_year);
   const size_t outputs_to_consider = rct_offsets.back() - (blocks_to_consider < rct_offsets.size() ? rct_offsets[rct_offsets.size() - blocks_to_consider - 1] : 0);
+
   begin = rct_offsets.data();
-  end = rct_offsets.data() + rct_offsets.size() - CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE;
-  num_rct_outputs = *(end - 1);
+  end = rct_offsets.data() + rct_offsets.size() - (std::max(1, CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE) - 1);
+
+  auto top_spendable_it =
+    std::next(
+            rct_offsets.rbegin(),
+            std::min<size_t>(
+                    rct_offsets.size(),
+                    std::max(1, CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE) - 1
+                )
+        );
+  if (top_spendable_it == rct_offsets.rend())
+    num_rct_outputs = 0;
+  else
+    num_rct_outputs = *top_spendable_it;
```

Comment:
```suggestion
num_rct_outputs = *(end - 1);
```

Now that we've changed the end iterator calculation, this is now correct again